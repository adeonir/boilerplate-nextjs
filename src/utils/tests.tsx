import { cleanup, render } from '@testing-library/react'
import type { ReactElement } from 'react'
import { afterEach } from 'vitest'

afterEach(() => {
  cleanup()
})

function customRender(ui: ReactElement, options = {}) {
  return render(ui, {
    wrapper: ({ children }) => children,
    ...options,
  })
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export { customRender as render }
