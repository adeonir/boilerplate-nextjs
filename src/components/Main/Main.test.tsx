import { render, screen } from '~/utils/tests'

import { Main } from '.'

const props = {
  title: 'NextJS Boilerplate',
  description:
    'NextJS, ReactJS, TypeScript, Styled-Components, Storybook, Jest and Testing-Library',
}

describe('<Main />', () => {
  it('renders the heading', () => {
    const { container } = render(<Main {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: props.description })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the colors correctly', () => {
    const { container } = render(<Main {...props} />)

    expect(container.firstChild).toHaveStyleRule('background-color', '#051330')
  })
})
