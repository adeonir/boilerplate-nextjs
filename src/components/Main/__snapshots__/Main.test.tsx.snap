// Vitest Snapshot v1

exports[`<Main /> > renders the heading 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #051330;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4rem;
  text-align: center;
}

.c1 {
  font-size: 2.5rem;
  margin-bottom: 1rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 300;
}

.c3 {
  margin-top: 4rem;
  width: min(40rem,100%);
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    NextJS Boilerplate
  </h1>
  <h2
    class="c2"
  >
    NextJS, ReactJS, TypeScript, Styled-Components, Storybook, Jest and Testing-Library
  </h2>
  <img
    alt="A developer walking and a screen with the text NextJS"
    class="c3"
    src="/assets/hero-illustration.svg"
  />
</main>
`;
